firmware-nucleo/smart-air-hockey-table/App/driver_display.c

```diff
@@ -104,28 +105,29 @@ void Driver_Display_Print(Display display, uint_fast8_t line, uint_fast8_t offse
 	if (offset * FONT_6X8_WIDTH >= DISPLAY_COLUMNS) return;
 
 	// Write format string to buffer
-	char buffer[2 * DISPLAY_COLUMNS / FONT_6X8_WIDTH]; // Extra space for escape codes
+	char buffer[DISPLAY_BUFFER_SIZE]; // Extra space for escape codes
 	va_list args;
 	va_start(args, format);
 	vsnprintf(buffer, sizeof(buffer), format, args);
 	va_end(args);
 
-	char* string = &buffer[0];
+	char* curr = &buffer[0];
 
 	// Send string to the display
 	bool inverted = false;
 	uint_fast8_t column = 0;
 	set_region(display, offset * FONT_6X8_WIDTH, DISPLAY_COLUMNS - 1, line, line);
 	transmit_start(display, true);
-	while (*string != '\0')
+	while (*curr != '\0' && curr - buffer < DISPLAY_BUFFER_SIZE)
 	{
-		uint8_t character = *(string++);
+		uint8_t character = *(curr++);
 		if (character == '\t')
 		{
 			inverted = !inverted;
 			continue;
 		}
-		if (character < FONT_6X8_MIN || character > FONT_6X8_MAX) character = ' ';
+		if (character < FONT_6X8_MIN || character > FONT_6X8_MAX)
+			character = ' ';

```

Comment:
I know it's not technically required, but,  if we are putting the instruction onto a new line, then shouldn't we add curly braces as well?

Reply:
done